Color/src/Graphics/Color/Space/DIN99.hs

```diff
@@ -0,0 +1,189 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE Strict #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UnboxedSums #-}
+{-# LANGUAGE UnboxedTuples #-}
+
+module Graphics.Color.Space.DIN99
+  ( pattern DIN99,
+    pattern ColorDIN,
+    DIN99,
+    deltaE,
+  )
+where
+
+import Foreign.Storable
+import GHC.Generics (Generic)
+import Graphics.Color.Illuminant.Wikipedia as W
+import Graphics.Color.Model.Internal
+import Graphics.Color.Space.CIE1976.LAB
+import Graphics.Color.Space.Internal
+
+data DIN99 (i :: k)
+
+-- | Color in DIN99 color space
+newtype instance Color (DIN99 i) e = DIN99 (V3 e)
+
+pattern ColorDIN :: e -> e -> e -> Color (DIN99 i) e
+pattern ColorDIN l' a' b' = DIN99 (V3 l' a' b')
+
+{-# COMPLETE ColorDIN #-}
+
+-- | CIE1976 `DIN99` color space
+deriving instance Eq e => Eq (Color (DIN99 i) e)
+
+-- | CIE1976 `DIN99` color space
+deriving instance Ord e => Ord (Color (DIN99 i) e)
+
+-- | CIE1976 `DIN99` color space
+deriving instance Functor (Color (DIN99 i))
+
+-- | CIE1976 `DIN99` color space
+deriving instance Applicative (Color (DIN99 i))
+
+-- | CIE1976 `DIN99` color space
+deriving instance Foldable (Color (DIN99 i))
+
+-- | CIE1976 `DIN99` color space
+deriving instance Traversable (Color (DIN99 i))
+
+-- | CIE1976 `DIN99` color space
+deriving instance Storable e => Storable (Color (DIN99 i) e)
+
+-- | CIE1976 `DIN99` color space
+instance (Illuminant i, Elevator e) => Show (Color (DIN99 i) e) where
+  showsPrec _ = showsColorModel
+
+instance (Illuminant i, Elevator e) => ColorModel (DIN99 i) e where
+  type Components (DIN99 i) e = (e, e, e)
+  toComponents (ColorDIN l' a' b') = (l', a', b')
+  {-# INLINE toComponents #-}
+  fromComponents (l', a', b') = ColorDIN l' a' b'
+  {-# INLINE fromComponents #-}
+
+instance (Illuminant i, Elevator e, RealFloat e) => ColorSpace (DIN99 (i :: k)) i e where
+  type BaseModel (DIN99 i) = DIN99 i
+  type BaseSpace (DIN99 i) = DIN99 i
+  toBaseSpace = id
+  {-# INLINE toBaseSpace #-}
+  fromBaseSpace = id
+  {-# INLINE fromBaseSpace #-}
+  luminance = luminance . dinToLAB DIN99Method
+  {-# INLINE luminance #-}
+  toColorXYZ = toColorXYZ . dinToLAB DIN99Method
+  {-# INLINE toColorXYZ #-}
+  fromColorXYZ = labToDIN DIN99Method . fromColorXYZ
+  {-# INLINE fromColorXYZ #-}
+
+kE :: RealFloat a => a
+kE = 1
+
+kCH :: RealFloat a => a
+kCH = 1
+
+data DINMethod
+  = ASTMD2244_07Method
+  | DIN99Method
+  | DIN99bMethod
+  | DIN99cMethod
+  | DIN99dMethod
+  deriving stock (Eq, Show, Read, Ord, Bounded, Enum, Generic)
```

Comment:
I don't think we need to enable `DerivingStrategies` extensions if all we are using is `stock`
```suggestion
  deriving (Eq, Show, Read, Ord, Bounded, Enum, Generic)
```